src/EditorFeatures/CSharp/TextStructureNavigation/CSharpTextStructureNavigatorProvider.cs

```diff
@@ -8,110 +8,117 @@
-    }
+        // We're in the middle.  Defer to the editor.  But make sure we don't go outside of the middle section.
+        var naturalExtent = naturalLanguageNavigator.GetExtentOfWord(position);
 
-    private static bool IsAtClosingQuote(SyntaxToken token, int position)
-        => token.Kind() switch
-        {
-            SyntaxKind.StringLiteralToken => position == token.Span.End - 1 && token.Text[^1] == '"',
-            SyntaxKind.Utf8StringLiteralToken => position == token.Span.End - 3 && token.Text is [.., '"', 'u' or 'U', '8'],
-            _ => throw ExceptionUtilities.Unreachable()
-        };
+        var intersection = naturalExtent.Span.Intersection(contentSpan.ToSpan());
+        return intersection is null ? naturalExtent : new TextExtent(intersection.Value, isSignificant: naturalExtent.IsSignificant);
 
-    protected override TextExtent GetExtentOfWordFromToken(SyntaxToken token, SnapshotPoint position)
-    {
-        if (token.Kind() is SyntaxKind.StringLiteralToken or SyntaxKind.Utf8StringLiteralToken &&
-            IsAtClosingQuote(token, position.Position))
-        {
-            // Special case to treat the closing quote of a string literal as a separate token.  This allows the
-            // cursor to stop during word navigation (Ctrl+LeftArrow, etc.) immediately before AND after the
-            // closing quote, just like it did in VS2013 and like it currently does for interpolated strings.
-            var span = new Span(position.Position, token.Span.End - position.Position);
-            return new TextExtent(new SnapshotSpan(position.Snapshot, span), isSignificant: true);
-        }
-        else if (token.Kind() is
-            SyntaxKind.SingleLineRawStringLiteralToken or
-            SyntaxKind.MultiLineRawStringLiteralToken or
-            SyntaxKind.Utf8SingleLineRawStringLiteralToken or
-            SyntaxKind.Utf8MultiLineRawStringLiteralToken)
-        {
-            var delimiterStart = GetStartOfRawStringLiteralEndDelimiter(token);
-            return new TextExtent(new SnapshotSpan(position.Snapshot, Span.FromBounds(delimiterStart, token.Span.End)), isSignificant: true);
-        }
-        else
+        (TextSpan startSpan, TextSpan contentSpan, TextSpan endSpan) GetStringLiteralParts()
         {
-            return base.GetExtentOfWordFromToken(token, position);
+            var start = token.Span.Start;
+            var contentStart = start;
+
+            if (CharAt(contentStart) == '@')
+                contentStart++;
+
+            if (CharAt(contentStart) == '"')
+                contentStart++;
+
+            if (isRawStringLiteral)
+            {
+                while (CharAt(contentStart) == '"')
+                    contentStart++;
+            }
+
+            var end = token.Span.End;
+            var contentEnd = end;
+
+            if (CharAt(contentEnd - 1) == '8')
+                contentEnd--;
+
+            if (CharAt(contentEnd - 1) is 'u' or 'U')
```

Comment:
>if (CharAt(contentEnd - 1) is 'u' or 'U') [](http://example.com/codeflow?start=12&length=41)

maybe combine this with the previous one since they are only valid in combination, right?